akka-actor/src/main/scala/akka/io/dns/internal/DnsClient.scala

```diff
@@ -111,6 +106,7 @@ import akka.pattern.{ BackoffOpts, BackoffSupervisor }
               case (s, _) =>
                 s ! Failure(new RuntimeException("Send failed to nameserver"))
                 inflightRequests -= msg.id
+                unstashQuestion(msg.id)
```

Comment:
Could just be `unstashAll`... the questions that still collide will still collide and get re-stashed.

Reply:
That's true